app/models/spree/payment_method/stripe_credit_card.rb

```diff
@@ -62,20 +79,32 @@ def create_profile(payment)
         }.merge! address_for(payment)
 
         source = update_source!(payment.source)
+
         if source.number.blank? && source.gateway_payment_profile_id.present?
-          creditcard = source.gateway_payment_profile_id
+          if v3_intents?
+            creditcard = ActiveMerchant::Billing::StripeGateway::StripePaymentToken.new('id' => source.gateway_payment_profile_id)
+          else
+            creditcard = source.gateway_payment_profile_id
+          end
         else
           creditcard = source
         end
 
         response = gateway.store(creditcard, options)
         if response.success?
-          payment.source.update_attributes!({
-            cc_type: payment.source.cc_type, # side-effect of update_source!
-            gateway_customer_profile_id: response.params['id'],
-            gateway_payment_profile_id: response.params['default_source'] || response.params['default_card']
-          })
-
+          if v3_intents?
+            payment.source.update_attributes!(
+              cc_type: payment.source.cc_type,
+              gateway_customer_profile_id: response.params['customer'],
+              gateway_payment_profile_id: response.params['id']
+            )
+          else
+            payment.source.update_attributes!(
+              cc_type: payment.source.cc_type, # side-effect of update_source!
```

Comment:
Can't really get the comment ` # side-effect of update_source!` here. Can you please expand a little bit?

Reply:
I don't know either, that comment was already there. If it has no significant meaning for you as well, probably it makes sense to just remove it.

Reply:
I doesn't tell me anything, I'd remove it.